### ISSUE TITLE
Rigidbody doesn't rotate when linear movement restriction on all axis is applied

### ISSUE THREAD

**Issue by Zuppor** [godotengine/godot#36847]
**Godot version:**
3.2 stable

**OS/device including version:**
Windows 10

**Issue description:**
rigidbody wont rotate when the linear movement on all the axis is locked

**Steps to reproduce:**
- make a rigidbody
- attach a collision shape as a child of the rigidbody and select a box shape
- adjust the collision shape extents and translation, so the parent pivot is on the side of it
- on the rigidbody, restrict the linear movement on all the axis, and restrict the angular movement on the x and z axis ([see image](https://imagebin.ca/v/5EfpCkLKgxlt))
- use a kinematic body as the player ([see image](https://imagebin.ca/v/5EfqteTAFOC5))
- play the scene and try to push the rigidbody with the player: the rigidbody should rotate around its y axis, but it doesn't, instead the player goes trough it
- now unlock the linear movement on one of the axis of the rigidbody
- play the scene and try to push again the rigidbody with the player: it should rotate around the y axis, and move along the unlocked axis (except if you unlocked the y axis)

**Minimal reproduction project:**

[IssueExample.zip](https://github.com/godotengine/godot/files/4297954/IssueExample.zip)

**Comment by Ev1lbl0w** [comment]
I believe I see your issue: you want to be able to "rotate" the door without it moving "vertically" on the y axis.

I have talked with a friend of mine who studies physics, and from what I understood, you need to always unlock the y axis if you wish to have angular velocity on that axis. So, if you want the door to rotate, you'd have to allow it to move in the Y axis *(even though it doesn't make much sense)*.

I don't know if what I said above is entirely true *(I'm no physics expert :stuck_out_tongue: )*, but tinkering with your project files and looking at the math, it seems like it.

**Comment by Zuppor** [comment]
I personally think it doesn't make sense. If I want to just rotate on the y axis without actually moving, why do I have to unlock the linear movement on that axis too?

**Comment by Ev1lbl0w** [comment]
I totally agree with you. However, I assume the code works something like this: when you lock the three axis, it means "lock all transformation on the X, Y and Z" axis. Which makes it's impossible to move and/or rotate a door, code-wise.

I think, for your specific case, you need to use an [HingeJoint](https://docs.godotengine.org/en/3.2/classes/class_hingejoint.html). Now, whether this specific physics scenario makes sense or not in the context of the engine, I have no clue ¯\_(ツ)_/¯

**Comment by Zuppor** [comment]
Yeah, I'm experimenting with the hingejoint already. 
But I have to say, I tried to do the same thing within another engine (Unity 3D) and it worked as I expected: the object was not able to move in any direction (as I locked the linear movement on all the axis), but was able to rotate along the y axis (with the angular movement locked on the x and z axis). I think it should be like that in the Godot engine, unless there is something I am ignoring and it works as intended? 🤔

**Comment by Ev1lbl0w** [comment]
@Zuppor a PR was merged (#38852) that might be related to your issue as well, maybe give it another shot?

**Comment by akien-mga** [comment]
CC @AndreaCatania to review if #38852 would address this.

**Comment by AndreaCatania** [comment]
Interesting use case; However, I'm not able to reproduce it in the Godot version 4: 
![ezgif com-video-to-gif(24)](https://user-images.githubusercontent.com/8342599/82354569-461a6080-9a01-11ea-89df-a5f861783eb9.gif)

I don't think the commit #38852 solves this problem (if confirmed) since it seems related to something other. I'm going to test it with version 3.2.

However, double check that the _door_ is not colliding with the ground. This may obstruct the movement.

**Comment by AndreaCatania** [comment]
I'm not able to reproduce it with the version 3.2 too. Here my sample project, hoping it will be useful to you:
[DoorTest.zip](https://github.com/godotengine/godot/files/4652004/DoorTest.zip)

**Comment by pouleyKetchoupp** [comment]
I can reproduce the issue with the original MRP in 3.2.4 beta 5.

Bullet: The player goes through the door.
Godot physics: The player collides with the door, but the door doesn't move.

**Comment by micimize** [comment]
To clarify/reconcile the above repros/vs fail to repo above (on 3.4.3), the demo Andrea posted of two colliding rigid bodies works, but programmed collision from a `KinematicBody` does not.

**Comment by Zireael07** [comment]
@micimize in that case it'll probably get closed, as infinite inertia is no longer a thing in Godot 4.0
